gnovm/pkg/gnolang/op_call.go

```diff
@@ -414,22 +366,32 @@ func (m *Machine) doOpPanic1() {
 }
 
 func (m *Machine) doOpPanic2() {
-	if len(m.Exceptions) == 0 {
+	if m.hasNoUnrecovered() {
+		m.Exceptions = nil
 		// Recovered from panic
 		m.PushOp(OpReturnFromBlock)
-		m.PushOp(OpReturnCallDefers)
-		m.PanicScope = 0
+		return
 	} else {
 		// Keep panicking
 		last := m.PopUntilLastCallFrame()
 		if last == nil {
 			// Build exception string just as go, separated by \n\t.
-			exs := make([]string, len(m.Exceptions))
+			var bld strings.Builder
```

Comment:
Can you preallocate some good amount to avoid multiple reallocations?
Maybe `bld.Grow(len(m.Exceptions) * 4)`